vcx/libvcx/src/issuer_credential.rs

```diff
@@ -392,28 +416,45 @@ pub fn encode_attributes(attributes: &str) -> Result<String, IssuerCredError> {
         }
     };
 
-    for (attr, vec) in map.iter_mut(){
-        let list = match vec.as_array_mut() {
-            Some(x) => x,
-            None => {
+    let mut dictionary = HashMap::new();
+
+    for (attr, attr_data) in map.iter_mut(){
+        let first_attr : &str = match attr_data {
+            // old style input such as {"address2":["101 Wilson Lane"]}
+            serde_json::Value::Array(array_type) => {
+
+                let attrib_value : &str = match array_type.get(0).and_then(serde_json::Value::as_str) {
```

Comment:
@glowkey @mattraffel should we handle array[2] as specific case when encoding already done by application level? Or may be we should accept only array of single element.

Reply:
agree